src/Http/Http.Extensions/src/RequestDelegateBuilder.cs

```diff
@@ -44,6 +46,11 @@ internal static class MapActionExpressionTreeBuilder
         private static readonly MemberExpression HttpResponseExpr = Expression.Property(HttpContextParameter, nameof(HttpContext.Response));
         private static readonly MemberExpression RequestAbortedExpr = Expression.Property(HttpContextParameter, nameof(HttpContext.RequestAborted));
 
+        /// <summary>
+        /// Builds a <see cref="RequestDelegate"/> implementation for <paramref name="action"/>.
+        /// </summary>
+        /// <param name="action">A request handler with any number of custom parameters that often produces a response with its return value.</param>
+        /// <returns>The <see cref="RequestDelegate"/></returns>
         public static RequestDelegate BuildRequestDelegate(Delegate action)
```

Comment:
I think this should take a `MethodInfo` instead of a delegate.

Reply:
How come?

Delegate argument seems more flexible. Change to MethodInfo ties using this to methods.

Reply:
Or can you always get a method using https://docs.microsoft.com/en-us/dotnet/api/system.delegate.method?view=net-5.0?

Reply:
After typing out the scenario there are 3 scenariosL
1. Delegate
2. MethodInfo with "this"
3. MethodInfo without "this"

MVC controller and the SignalR are number 2. We don't build any frameworks with number 3 today. That would lead to 3 overloads of this method:

```C#
public sealed class RequestDelegateBuilder 
{
    public static RequestDelegate Build(Delegate requestHandler);
    public static RequestDelegate Build(MethodInfo methodInfo, Func<HttpContext, object> targetFactory);
    public static RequestDelegate Build(MethodInfo methodInfo);
}
```

1.

```C#
Func<string> hello = () => "Hello World";
RequestDelegate rd1 = RequestDelegateBuilder.Build(hello);
app.Run(rd1);

RequestDelegate rd2 = RequestDelegateBuilder.Build((Func<string>)MyClass.Hello);
app.Run(rd2);

public class MyClass
{
    public static string Hello() => "Hello World";
}
```

2.

```C#
MethodInfo methodInfo = typeof(MyClass).GetMethod(nameof(MyClass.Hello));

// This will create an instance of MyClass per request, giving the factory a chance to construct
// MyClass using per request scoped objects
RequestDelegate rd1 = RequestDelegateBuilder.Build(methodInfo, context => new MyClass());

// Example with DI
RequestDelegate rd2 = RequestDelegateBuilder.Build(methodInfo, context => context.RequestServices.GetRequiredService<MyClass>());

app.Run(rd1);
app.Run(rd2);

public class MyClass
{
    public string Hello() => "Hello World";
}
```

3.

```C#
MethodInfo methodInfo = typeof(MyClass).GetMethod(nameof(MyClass.Hello));

RequestDelegate rd = RequestDelegateBuilder.Build(methodInfo);

app.Run(rd);

public class MyClass
{
    public static string Hello() => "Hello World";
}
```

This is basically the same as 2, with a null targetFactory.